### PR TITLE
fix: Allow ArgoCD CR to override keycloak resource request/limit defaults

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind bug

**What does this PR do / why we need it**:
Allow ArgoCD CR to override keycloak resource request/limit defaults

**How to test changes / Special notes to the reviewer**:
1. Run/Install the operator
2. Create ArgoCD instance with Keycloak (.spec.sso.provider = keycloak)
3. Update the SSO resource request/limit
4. Verify the changes

### CODE REVIEW

**Comment by wtam2018** [comment]
@iam-veeramalla please look at the failed tests.

**Comment by iam-veeramalla** [comment]
> @iam-veeramalla please look at the failed tests.

@wtam2018 on it. I have to refactor the code slightly fix the failing test. I will verify the new changes and raise a new PR

File: pkg/controller/argocd/sso.go [hunk 1]
```diff
@@ -133,47 +134,73 @@ func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
...
+
+					// Create a keycloak realm and publish.
+					response, err := createRealm(cfg)
 					if err != nil {
-						log.Error(err, fmt.Sprintf("Failed to update OIDC Configuration for ArgoCD %s in namespace %s",
+						log.Error(err, fmt.Sprintf("Failed posting keycloak realm configuration for ArgoCD %s in namespace %s",
 							cr.Name, cr.Namespace))
 						return err
 					}
+
```
Comment (iam-veeramalla) [inline pkg/controller/argocd/sso.go:183]: There is no change to this block of code.

File: pkg/controller/argocd/sso.go [hunk 2]
```diff
@@ -133,47 +134,73 @@ func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
 					Namespace: cr.Namespace,
 				},
 			}
-			err = r.client.Get(context.TODO(), types.NamespacedName{Name: existingDC.Name, Namespace: existingDC.Namespace}, existingDC)
-			if err != nil {
-				log.Error(err, fmt.Sprintf("Keycloak Deployment not found or being created for ArgoCD %s in namespace %s",
-					cr.Name, cr.Namespace))
-			}
 
-			// If Keycloak deployment exists and a realm is already created for ArgoCD, Do not create a new one.
```
Comment (iam-veeramalla) [inline pkg/controller/argocd/sso.go:142]: There is no change to this block of code

File: pkg/controller/argocd/sso.go [hunk 3]
```diff
@@ -133,47 +134,73 @@ func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
...
-				keycloakRouteURL := cfg.KeycloakURL
+				// Check if the Image is updated by the user.
+				existingImage := existingDC.Spec.Template.Spec.Containers[0].Image
+				desiredImage := getKeycloakContainerImage(cr)
+				if existingImage != desiredImage {
+					existingDC.Spec.Template.Spec.Containers[0].Image = desiredImage
+					changed = true
+				}
 
-				// Create a keycloak realm and publish.
```
Comment (iam-veeramalla) [inline pkg/controller/argocd/sso.go:154]: There is no change to this block of code

**Review by wtam2018 — APPROVED** [review]
LGTM
thanks!

**Pull request merged**